presto-main/src/test/java/com/facebook/presto/operator/scalar/TestZipWithFunction.java

```diff
@@ -107,4 +108,12 @@ public void testWithNull()
                 new ArrayType(UNKNOWN),
                 asList(null, null));
     }
+
+    @Test
+    public void testWrongParameterCounts()
+            throws Exception
+    {
+        assertInvalidFunction("zip_with(ARRAY[1], ARRAY['a'], x -> x)", INVALID_PARAMETER_USAGE, "expected a lambda function takes 2 parameters (actual: 1)");
+        assertInvalidFunction("zip_with(ARRAY[1], ARRAY['a'], (x, y, z) -> (x, y, z))", INVALID_PARAMETER_USAGE, "expected a lambda function takes 2 parameters (actual: 3)");
```

Comment:
You could have a test that verifies error location is correctly reported. (I mean "line x:y" as in `com.facebook.presto.sql.analyzer.TestAnalyzer#testGroupByWithSubquerySelectExpression`)